### PR TITLE
cleaned up package structure, fixed some api's and checkstyles

### PR DESCRIPTION
**Pull request by lwahlmeier**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 group = org.threadly
 version = 0.1.0-SNAPSHOT
-threadlyVersion = 3.8.0
-litesocketsVersion = 2.0.0
+threadlyVersion = 4.1.0
```
Comment (jentfoo) [inline gradle.properties]: In theory threadly should not need to be specified, it should be picked up from litesockets.

File: src/main/java/org/threadly/litesockets/protocol/http/structures/HTTPAddress.java [hunk 2]
```diff
@@ -1,5 +1,10 @@
-package org.threadly.litesockets.protocol.http;
+package org.threadly.litesockets.protocol.http.structures;
 
+/**
+ * Simple Immutable class that is used for http Connections.  Its mainly used
+ * to know if 2 address will connect to the exact same place or not. 
+ *
+ */
 public class HTTPAddress {
```
Comment (jentfoo) [inline src/main/java/org/threadly/litesockets/protocol/http/structures/HTTPAddress.java]: Why not use an InetSocketAddress?  Only information that would be missing is the "doSSL" which seems unrelated to the address.

**Comment by jentfoo** [comment]
I made a couple small comments, but otherwise this looks good to me.  Is this still a WIP, or is it ready to merge?  Do you want an artifact after it's merged?

File: src/main/java/org/threadly/litesockets/protocol/http/structures/HTTPAddress.java (hunk 2 above)
Comment (lwahlmeier) [inline src/main/java/org/threadly/litesockets/protocol/http/structures/HTTPAddress.java]: Its not, I need a way to know if a connection can be reused or not, so I need a compairable of host/port/ssl to do that.

File: gradle.properties (hunk 1 above)
Comment (lwahlmeier) [inline gradle.properties]: It would, but its a direct dependency, so I like to keep it in the dependencies because of that.

**Comment by lwahlmeier** [comment]
yeah its still a wip, I have a bug fix I still need to get into this merge, and we should talk about the HTTPClient object I think it needs to change quite a bit.

**Comment by lwahlmeier** [comment]
@jentfoo Sorry this took me so long, most of the changes are in HTTPClient.  I needed away to use it with tons of HTTP connections at once using a Threaded SocketExecuter.  Let me know if you have any questions and merge when your happy with it

**Comment by jentfoo** [comment]
LGTM
